src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -63,7 +64,22 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
     // needed, because studentExam is downloaded only when exam is started
     exam: Exam;
     examTitle = '';
-    studentExam: StudentExam;
+    studentExamWithGrade: StudentExamWithGradeDTO;
```

Comment:
I don't understand and don't like this change. This is a really critical component, we should not change it!
Why do we need a DTO here?

Reply:
The changes below are also problematic, because studentExam is access from html.
I would prefer, if we can get rid of this change here and instead load the grade information in the `jhi-exam-participation-summary`. This will at least not influence the exam participation in any way

Reply:
The server endpoint for `loadStudentExamWithExercisesForSummary` is modified to return `StudentExamWithGradeDTO` instead of `StudentExam`. `StudentExamWithGradeDTO` wraps the old `StudentExam` as-is and just adds extra fields. The aim here is to include the calculated points and grade (if applicable and present) in the same response so we send one less request to the server.

At the marked line above, we store the new `StudentExamWithGradeDTO` just for passing it to `jhi-exam-participation-summary` in the html. For other cases, I tried to keep the code changes to a minimum in order to not break existing functionality and added the getter and setter for `studentExam` so that after the [subscribe callback](https://github.com/ls1intum/Artemis/pull/4932/files/b9006d604449be6fb79c9406b06b0ac31ad92513#diff-bf0b46e59bcde52e27fb94dbfe6f585078130cb1e0e61410504db0a2f871cc11R181-R183) is executed, the remaining code can work as before.

> The changes below are also problematic, because studentExam is access from html.

Is it about Angular's optimization for field accesses vs method calls in html? If yes I can convert `studentExam` to a concrete field and set it [here](https://github.com/ls1intum/Artemis/pull/4932/files/b9006d604449be6fb79c9406b06b0ac31ad92513#diff-bf0b46e59bcde52e27fb94dbfe6f585078130cb1e0e61410504db0a2f871cc11R182) without the changes I wrote in the paragraph below.

> instead load the grade information in the jhi-exam-participation-summary

My hesitation for this is re-introducing the extra request that I removed, but if you think it will be better then I can modify the `ExamParticipationService.loadStudentExamWithExercisesForSummary` method so that it extracts the `StudentExam` from `StudentExamWithGradeDTO` so `exam-participation.component.ts` is untouched. Then I would need to modify `jhi-exam-participation-summary` to get `StudentExamWithGradeDTO` from either the input if it is given or by sending another request to the server if it is not given.